blog/2024-11-27-flet-v-0-25-release-announcement.md

```diff
@@ -0,0 +1,294 @@
+```
+
+or the same in `pyproject.toml` (read about `pyproject.toml` support below):
+
+```toml
+[tool.flet.ios.info] # --info-plist
+NSCameraUsageDescription = "This app uses the camera to ..."
+```
+
+An example of setting Android permissions and features:
+
+```
+flet build \
+    --android-permissions android.permission.READ_EXTERNAL_STORAGE=True \
+      android.permission.WRITE_EXTERNAL_STORAGE=True \
+    --android-features android.hardware.location.network=False
+```
+
+[Read more about permissions in the docs](/docs/publish#permissions).
+
+### Control over app compilation and cleanup
+
+`flet build` command is no longer compiling app `.py` files into `.pyc` by default which allows you to defer discovery of any syntax errors in your app and complete the packaging.
+
+You can control the compilation and cleanup with the following new options:
+
+* `--compile-app` - compile app's `.py` files.
+* `--compile-packages` - compile installed packages' `.py` files.
+* `--cleanup-on-compile` - remove unnecessary files upon successful compilation.
+
+### Signing Android bundles
+
+We also added new options for signing Android builds:
+
+* `--android-signing-key-store` - path to an upload keystore `.jks` file for Android apps.
+* `--android-signing-key-store-password` - Android signing store password.
+* `--android-signing-key-alias` - Android signing key alias. Default is "upload".
+* `--android-signing-key-password` - Android signing key password.
+
+Read [Build and release an Android app](https://docs.flutter.dev/deployment/android#signing-the-app) for more information on how to configure upload key for Android builds.
+
+### Deep linking configuration
+
+There is a new `--deep-linking-url` option to configure deep linking for iOS and Android builds. The value must be in the format `<sheme>://<host>`.
+
+### Faster re-builds
+
+Ephemeral Flutter app created by `flet build` command is no longer being re-created on every build in a temp directory, but cached in `build/flutter` directory which gives faster re-builds, improves packaging troubleshooting and does not pollute user temp directory.
+
+### Split APKs per ABI
+
+`flet build` now provides the built-in `--split-per-abi` option to split the APKs per ABIs.
+
+## "Data" and "Temp" directories for the app
+
+Flet developers have been asking where to store application data, such as uploaded files, SQLite databases, etc. that are persistent across application updates.
+
+This release introduce two environment variables that are available in your Flet apps:
+
+* `FLET_APP_DATA` - directory for storing application data that is preserved between app updates. That directory is already pre-created and its location depends on the platform the app is running on.
+* `FLET_APP_TEMP` - directory for temporary application files, i.e. cache. That directory is already pre-created and its location depends on the platform the app is running on.
```

Comment:
Must not happen, but I think it will be nice to have "storage" in their names similar to their paths.
```suggestion
* `FLET_APP_STORAGE_DATA` - directory for storing application data that is preserved between app updates. That directory is already pre-created and its location depends on the platform the app is running on.
* `FLET_APP_STORAGE_TEMP` - directory for temporary application files, i.e. cache. That directory is already pre-created and its location depends on the platform the app is running on.
```